README.md

```diff
@@ -1,7 +1,334 @@
+```csharp
+PushDocument document = new PushDocument("http://www.coveo.com/secured") {
+    ClickableUri = "http://www.coveo.com/secured",
+    ModifiedDate = DateTime.UtcNow
+};
+
+// Push the document with the security associated to it.
+PermissionIdentity user = new PermissionIdentity(@"acme\johndoe", PermissionIdentityType.User);
+PermissionIdentity group = new PermissionIdentity(@"acme\team", PermissionIdentityType.Group);
+
+document.SetSimpleAllowedAndDeniedPermissions(new List<PermissionIdentity>{ user, group }, new List<PermissionIdentity>());
+
+client.DocumentManager.AddOrUpdateDocument(sourceId, document, null);
+
+// Push the permission mapping for a user from Active Directory to email.
+PermissionIdentityBody userBody = new PermissionIdentityBody(user);
+userBody.Mappings.Add(new PermissionIdentity("johndoe@acme.com", PermissionIdentityType.User));
+
+// Push the permission mapping for a group from Active Directory to email.
+PermissionIdentityBody groupBody = new PermissionIdentityBody(group);
+groupBody.Mappings.Add(new PermissionIdentity("teammember1@team.acme.com", PermissionIdentityType.User));
+groupBody.Mappings.Add(new PermissionIdentity("teammember2@team.acme.com", PermissionIdentityType.User));
+
+client.PermissionManager.AddOrUpdateIdentity(expandedProviderId, null, userBody);
+client.PermissionManager.AddOrUpdateIdentity(expandedProviderId, null, groupBody);
+```
+
+### Push identities in batch
+For overall performance, it is better to push your identities in batches.
+```csharp
+PermissionIdentity member1 = new PermissionIdentity(@"acme\member1", PermissionIdentityType.User);
+PermissionIdentity member2 = new PermissionIdentity(@"acme\member2", PermissionIdentityType.User);
+
+PermissionIdentityBody bodyMember1 = new PermissionIdentityBody(member1);
+bodyMember1.Mappings.Add(new PermissionIdentity("member1@acme.com", PermissionIdentityType.User));
+
+PermissionIdentityBody bodyMember2 = new PermissionIdentityBody(member2);
+bodyMember2.Mappings.Add(new PermissionIdentity("member2@acme.com", PermissionIdentityType.User));
+
+List<PermissionIdentityBody> mappingsMembersToAddOrUpdate = new List<PermissionIdentityBody> {
+    bodyMember1,
+    bodyMember2
+};
+
+List<PermissionIdentityBody> membersToAddOrUpdate = new List<PermissionIdentityBody> {
+    new PermissionIdentityBody(new PermissionIdentity(@"acme\team2", PermissionIdentityType.Group)) {
+        Members = new List<PermissionIdentity> {
+            member1,
+            member2
+        }
+    }
+};
+
+// Push user mappings and group members in one batched operation.
+client.PermissionManager.AddOrUpdateIdentities(expandedProviderId, null, mappingsMembersToAddOrUpdate.Concat(membersToAddOrUpdate).ToList());
+```
+
+### Disable a single identity
```

Comment:
It's ok to write "identity" rather than always "security identity", but we should be precise the first time. So:
"Disable a single identity" -> "Disable a single security identity"

Reply:
```suggestion
### Disable a single security identity
```